avocado/utils/asset.py

```diff
@@ -69,6 +69,28 @@ def __init__(self, name, asset_hash, algorithm, locations, cache_dirs,
         self.basename = os.path.basename(self.nameobj.path)
         self.expire = expire
 
+    def _get_writable_cache_dir(self):
+        """
+        Returns the first available writable cache directory
+
+        When a asset has to be downloaded, a writable cache directory
+        is then needed. The first available writable cache directory
+        will be used.
+
+        :returns: the first writable cache dir
+        :rtype: str
+        :raises: EnvironmentError
+        """
+        result = None
+        for cache_dir in self.cache_dirs:
+            cache_dir = os.path.expanduser(cache_dir)
+            if utils_path.usable_rw_dir(cache_dir):
+                result = cache_dir
+                break
+        if result is None:
```

Comment:
How about a simple speedup by using `return` instead of `break` and `raise` directly after the `for` loop?

Reply:
True, I guess I didn't thought of that.